src/shims/unix/fd.rs

```diff
@@ -257,6 +257,135 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         interp_ok(())
     }
 
+    fn readv(
+        &mut self,
+        fd_num: i32,
+        iov_ptr: &OpTy<'tcx>,
+        iovcnt: i32,
+        offset: Option<i128>,
+        dest: &MPlaceTy<'tcx>,
+    ) -> InterpResult<'tcx> {
+        let this = self.eval_context_mut();
+
+        // Early returns for empty or invalid cases
+        if iovcnt == 0 {
```

Comment:
Is this justified by the docs? Also, what does the "or invalid" mean here?

Reply:
* I've updated the documentation to clarify the behavior of readv when iovcnt is 0. The updated documentation now explicitly covers:

    1. The function's return value semantics based on the POSIX specification
    2. The meaning of various error conditions and their corresponding return codes
    3. Special cases like empty buffer lists (iovcnt = 0)

Reply:
I wasn't asking about your documentation, I was asking about the official documentation for `readv`. Often in cases like this we quite a sentence from the docs, and link to the docs, to explain why particular corner cases are handled the way they are.